### PR TITLE
Add prometheus metric for number of name entries deleted by fqdn gc

### PR DESCRIPTION
**Pull request by h3llix**
Fixes:#16305

Signed-off-by: Gaurav Genani <h3llix.pvt@gmail.com>

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -1098,10 +1103,18 @@ func CreateConfiguration(metricsEnabled []string) (Configuration, []prometheus.C
...
 				Help:      "Number of FQDNs that have been cleaned on FQDN Garbage collector job",
 			})
-
 			collectors = append(collectors, FQDNGarbageCollectorCleanedTotal)
 			c.FQDNGarbageCollectorCleanedTotalEnabled = true
 
+		case Namespace + "_fqdn_gc_deleted_name_entries":
+			FQDNGarbageCollectorDeletedEntriesTotal = prometheus.NewCounter(prometheus.CounterOpts{
+				Namespace: Namespace,
+				Name:      "_fqdn_gc_deleted_name_entries",
```
Comment (aanm) [inline pkg/metrics/metrics.go]: remove the `_` from the `_fqdn`

File: daemon/cmd/fqdn.go [hunk 2]
```diff
@@ -234,11 +234,10 @@ func (d *Daemon) bootstrapFQDN(possibleEndpoints map[uint16]*endpoint.Endpoint,
 			namesCount := len(namesToClean)
 			// Limit the amount of info level logging to some sane amount
 			if namesCount > 20 {
-				// namedsToClean is only used for logging after this so we can reslice it in place
+				// namesToClean is only used for logging a	fter this so we can reslice it in place
 				namesToClean = namesToClean[:20]
 			}
```
Comment (aanm) [inline daemon/cmd/fqdn.go]: This can now be removed entirely

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/cmd/fqdn.go [hunk 3]
```diff
@@ -231,14 +231,7 @@ func (d *Daemon) bootstrapFQDN(possibleEndpoints map[uint16]*endpoint.Endpoint,
...
 			_, err := d.dnsNameManager.ForceGenerateDNS(context.TODO(), namesToClean)
-			namesCount := len(namesToClean)
-			// Limit the amount of info level logging to some sane amount
-			if namesCount > 20 {
-				// namedsToClean is only used for logging after this so we can reslice it in place
-				namesToClean = namesToClean[:20]
-			}
-			log.WithField(logfields.Controller, dnsGCJobName).Infof(
-				"FQDN garbage collector work deleted %d name entries: %s", namesCount, strings.Join(namesToClean, ","))
+			metrics.FQDNGarbageCollectorDeletedEntriesTotal.Add(float64(len(namesToClean)))
```
Comment (aanm) [inline daemon/cmd/fqdn.go:234]: Wait a second, it turns out the metrics existed already, (line 232). So in reality we only need to remove the log lines? But it looks like @jrajahalme recently added this. Pinging @jrajahalme to double check if it's really useful to keep this message bring printed.

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/cmd/fqdn.go (hunk 3 above)
Comment (h3llix) [reply daemon/cmd/fqdn.go:234]: Yes . I missed it , looks like line 232 does the same thing

**Comment by jrajahalme** [comment]
This logging was added explicitly to get visibility to the actual names being deleted (bounded to some sane number to limit the logging). The need was for debugging purposes on customer environments that can't run with `--debug`. I wonder if it is possible to pass the deleted names to Prometheus somehow instead?

**Comment by jrajahalme** [comment]
The situation where this logging would be helpful is when `toFQDNs` policies unexpectedly drop traffic that is allowed. We have fixed couple of bugs related to this, but if any are left then this logging showing the actual names would be useful. I recall that there are no current incidents where this is happening, though.

**Review by jrajahalme — COMMENTED** [review]
@aanm Maybe you can decide what to do based on my comments above?

**Comment by aanm** [comment]
>This logging was added explicitly to get visibility to the actual names being deleted

@jrajahalme couldn't we derive that information from the sysdump by keeping the last ~100 FQDNs names deleted and when they were deleted? Keeping a map with ~100 entries sounds better than filling up the logs with that information.

>I wonder if it is possible to pass the deleted names to Prometheus somehow instead?

Technically it is possible but we shouldn't do it as it will create a lot of label cardinality and will not scale.

**Comment by jrajahalme** [comment]
> @jrajahalme couldn't we derive that information from the sysdump by keeping the last ~100 FQDNs names deleted and when they were deleted? Keeping a map with ~100 entries sounds better than filling up the logs with that information.
> 

So presumably upon a problem a sysdump would be taken soon enough for the 100 names to roll over? Sounds like a plan, quite a bit of work though!

**Comment by aanm** [comment]
@h3llix I've changed the issue description a little bit given the above discussion so I don't think this PR can fix the issue as it is.

**Pull request closed without merging**